frontend/src/util/apiUtil.js

```diff
@@ -0,0 +1,51 @@
+function setHeaders(headers) {
+  
+  if (localStorage.jwtToken) {
+    return {
+      ...headers,
+      Authorization: localStorage.jwtToken,
+    };
+  } else {
+    return headers;
+  }
+}
+
+function genericFetchRequset(type, endpoint, data = {}, onSuccess, onError) {
+  debugger
+    let params = {
+      method: type,
+      headers: setHeaders({
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      }),
+      body: data,
+    };
+
+    if (type === "GET") {
+      debugger
+      delete params.body
+    } else if ((type === "POST" || type === "PUT") && (data.isHidden || data.email)) {
```

Comment:
Why not always do the stringify? the data should always be a json. why condition it?

Reply:
Because if there is a file I use formData

Reply:
I don't understand the condition - why check for email? why not just check if there's file or not?

Reply:
I check if it's regular data by checking if there's an isHidden field but the login form doesn't have an isHidden field